src/common/buffer.cc

```diff
@@ -125,10 +128,12 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
     if (r < 0)
       return r;
     buf[r] = '\0';
-    size_t size = strict_strtol(buf, 10, &err);
-    if (!err.empty())
-      return -EIO;
-    buffer_max_pipe_size.set(size);
+    errno = 0;
+    size_t size = strtoul(buf, nullptr, 10);
+    if (errno == 0)
```

Comment:
@tchaikov Thank you very much. It is always pleasure to learn from more experienced than you are. I will fix accordingly.